docs/server-options.md

```diff
@@ -0,0 +1,111 @@
+---
+id: server-options
+title: Temporal server options
+sidebar_label: Start options
+---
+
+## Overview
+
+To run the Temporal, include the temporal server package `go.temporal.io/server/temporal` in your Go application. Then use it to create and start a new server:
+
+```go
+s := temporal.NewServer()
+err := s.Start()
+if err != nil{
+	log.Fatal(err)
+}
+```
+
+## Options
+
+`NewServer()` accepts functions as parameters. Each function returns a `ServerOption` that is applied to the instance.
+
+### WithConfig
+
+Overrides the default configuration with a custom configuration that is defined in the config package `go.temporal.io/server/common/service/config`.
+
+```go
+s := tempora.NewServer(
+	temporal.WithConfig(cfg),
+)
+```
+
+### WithConfigLoader
+
+Load a custom configuration from a file.
+
+```go
+s := tempora.NewServer(
+	temporal.WithConfigLoader(configDir, env, zone),
+)
+```
+
+### ForServices
+
+Sets the list of all valid temporal services. The default can be used from the `go.temporal.io/server/temporal` package.
+
+```go
+s := tempora.NewServer(
+	temporal.ForServices(temporal.Services),
+)
+```
+
+### InterruptOn
+
+This option provides a channel that interrupts the server on the signal from that channel.
```

Comment:
maybe spell out usecase - when is this helpful? is this a rare usecase?

Reply:
For these server options, I will need to get more input from the team.
In case I don't get this info in a timely manner, I created a task in our board to track.

Reply:
wonderful